### PR TITLE
Prevent unnecessary joins / complex conditions in delete

### PR DESCRIPTION
**Pull request by mvdriel**
Prevent unnecessary joins / complex conditions in delete when there are boolean operators in the conditions.

### CODE REVIEW

File: lib/Cake/Model/Datasource/DboSource.php [hunk 1]
```diff
@@ -184,6 +184,13 @@ class DboSource extends DataSource {
 	protected $_sqlOps = array('like', 'ilike', 'rlike', 'or', 'not', 'in', 'between', 'regexp', 'similar to');
 
 /**
+ * The set of valid SQL boolean operations usable in a WHERE statement
+ *
+ * @var array
+ */
+	protected $_sqlBoolOps = array('and', 'or', 'not', 'and not', 'or not', 'xor', '||', '&&');
```
Comment (markstory) [inline lib/Cake/Model/Datasource/DboSource.php]: `||` is a concat operator in postgres. I'm not sure it and `&&` need to be here.

Comment (markstory) [inline lib/Cake/Model/Datasource/DboSource.php]: Nevermind, this is just lifting a buried var.

**Comment by markstory** [comment]
Many of the travis builds failed for this pull request. Do the tests pass locally for you?

**Comment by mvdriel** [comment]
@markstory Oops, there was a line missing in my commit. Tests should pass now.

**Comment by markstory** [comment]
Thanks @mvdriel :smile:

**Pull request merged**